File: frontend/src/model/model/entity/SiteTag.ts
import BaseEntity from '../base/BaseEntity.ts'
import { notNullish } from '@renderer/utils/CommonUtil.ts'

/**
 * 站点标签
 */
export default class SiteTag extends BaseEntity {
  /**
   * 标签来源站点id
   */
  siteId: number | undefined | null
  /**
   * 站点中标签的id
   */
  siteTagId: string | undefined | null
  /**
   * 站点中标签的名称
   */
  siteTagName: string | undefined | null
  /**
   * 上级标签id
   */
  baseSiteTagId: string | undefined | null
  /**
   * 描述
   */
  description: string | undefined | null
  /**
   * 站点标签对应的本地标签id
   */
  localTagId: number | undefined | null
  /**
   * 最后一次使用的时间
   */
  lastUse: number | null | undefined

  constructor(siteTag?: SiteTag) {
    super(siteTag)
    if (notNullish(siteTag)) {
      this.siteId = siteTag.siteId
      this.siteTagId = siteTag.siteTagId
      this.siteTagName = siteTag.siteTagName
      this.baseSiteTagId = siteTag.baseSiteTagId
      this.description = siteTag.description
      this.localTagId = siteTag.localTagId
      this.lastUse = siteTag.lastUse
    }
  }
}

File: backend/base/model/entity/site_tag.go
package entity

import (
	"database/sql"

	"github.com/library-squirrel/backend/base/model"
)

// SiteTag 站点标签
type SiteTag struct {
	*model.BaseEntity
	SiteID        sql.NullInt64  `gorm:"column:site_id;uniqueIndex:idx_site_tag_site_site_tag" json:"siteId"`
	SiteTagID     sql.NullString `gorm:"column:site_tag_id;uniqueIndex:idx_site_tag_site_site_tag" json:"siteTagId"`
	SiteTagName   sql.NullString `gorm:"column:site_tag_name" json:"siteTagName"`
	BaseSiteTagID sql.NullString `gorm:"column:base_site_tag_id" json:"baseSiteTagId"`
	Description   sql.NullString `gorm:"column:description" json:"description"`
	LocalTagID    sql.NullInt64  `gorm:"column:local_tag_id" json:"localTagId"`
	Namespace     sql.NullString `gorm:"column:namespace" json:"namespace"` // 站点侧 namespace（language/character/parody/female/male/misc/general 等）；null=站点无 namespace（pixiv 等）
	LastUse       sql.NullInt64  `gorm:"column:last_use" json:"lastUse"`
}

func NewSiteTag() *SiteTag {
	return &SiteTag{
		BaseEntity: &model.BaseEntity{},
	}
}

func (SiteTag) TableName() string {
	return "site_tag"
}
